pkg/pillar/types/zboottypes.go

```diff
@@ -11,10 +16,54 @@ type ZbootConfig struct {
 }
 
 // Key returns the key used in pubsub for ZbootConfig
-func (status ZbootConfig) Key() string {
+func (config ZbootConfig) Key() string {
 	return status.PartitionLabel
 }
 
+// LogCreate :
+func (config ZbootConfig) LogCreate() {
+	logObject := base.NewLogObject(base.ZbootConfigLogType, "",
+		nilUUID, config.LogKey())
+	if logObject == nil {
+		return
+	}
+	logObject.CloneAndAddField("test-complete", config.TestComplete).
```

Comment:
The current object field keys that we already have in EVE, I plan to add them to the devicevX-* template index that we pass to elastic search. That should force the types for current keys in ES.
For the new field keys that we add in now/future, I think it makes it easier if we suffix the key values with their supposed types. For ex: test-complete becomes test-complete-bool, partition-state becomes partition-state-str etc.

This way we can add dynamic mappings that can match on names (rudimentary regex) and force a type on the value in ES.
ex:
```
        "strings": {
          "match: "*-str"
          "mapping": {
           "type": "text",
           "fields": {
             "keyword": {
               "type": "keyword"
             }
           }
         }
        },
        "integers": {
          "match: "*-int",
          "mapping": {
           "type": "integer",
         }
        },
        "bools": {
          "match: "*-bool",
          "mapping": {
           "type": "bool",
         }
        }
```
If we do not match anything that we know, the default last dynamic-mapping can be forced to "text".

Reply:
> For the new field keys that we add in now/future, I think it makes it easier if we suffix the key values with their supposed types. For ex: test-complete becomes test-complete-bool, partition-state becomes partition-state-str etc.

Good idea. I'll do that in this PR for the new ones. Can we have string be the default to avoid adding -str? And do we need to make a difference between int and uint?

I assume that as we replicate existing datatypes (such as SwState) to be logged for additional structures we can keep their names (such as "state"). We should do that for downloader and verifier types soon.

Reply:
Yes, we can keep that existing names and use them in other objects also. Sure, default can be made string to avoid having to suffix -str.
Also, I think it would be easier to suffix any number type with "-num" and then treat them as "long" in suffix. What do you think?